### PR TITLE
DX: CI cleanup

### PR DESCRIPTION
**Pull request by kubawerlos**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -96,7 +96,7 @@ jobs:
           max_attempts: 5
           retry_wait_seconds: 30
           command: |
-            composer update --optimize-autoloader --no-interaction --no-progress ${{ matrix.composer-flags }}
+            composer update --no-interaction --no-progress --optimize-autoloader ${{ matrix.composer-flags }}
```
Comment (keradus) [inline .github/workflows/ci.yml:99]: what's the difference?

File: .github/workflows/sca.yml [hunk 2]
```diff
@@ -66,7 +66,7 @@ jobs:
         run: |
           ./dev-tools/check_file_permissions.sh
           ./dev-tools/check_trailing_spaces.sh
-          ./dev-tools/vendor/bin/phpstan analyse
+          ./dev-tools/vendor/bin/phpstan analyse --no-progress
```
Comment (keradus) [inline .github/workflows/sca.yml:69]: why?
we do progress for other tools, eg for PHP CS Fixer itself

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kubawerlos) [reply .github/workflows/ci.yml:99]: None, only sorted alphabetically (as I was trying different things before squashing).

File: .github/workflows/sca.yml (hunk 2 above)
Comment (kubawerlos) [reply .github/workflows/sca.yml:69]: To not have this "beautiful" logs:
image
AFAIK other tools [don't](https://pipelines.actions.githubusercontent.com/6HzPVZnYJGwfOotYxu8ZJJIcqdtpCW1gc7jXnYZ3NyGFf1CXiD/_apis/pipelines/1/runs/857/signedlogcontent/3?urlExpires=2020-12-16T21%3A51%3A21.1008647Z&urlSigningMethod=HMACV1&urlSignature=MsjzJk%2FEv2KQE80npXNeKmj7CpAw6s4QyXn1XUBkaeI%3D) have such progress.

Comment (keradus) [reply .github/workflows/sca.yml:69]: > Uri expired

composer checker doesn't have it
phpmd doesn't have it, still (we are longly awaiting for it, the tool looks to froze)
php-cs-fixer we run with `-v` for progress

i would keep the progress indicator to show sth is happening for non-instant tools, eg phpstan

what's the downside of having it? if it's green, it's getting collapsed.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (keradus) [reply .github/workflows/ci.yml:99]: i don't see value of changing the order, and nobody will manually maintain the alphabetical order of cli params.

File: .github/workflows/sca.yml (hunk 2 above)
Comment (kubawerlos) [reply .github/workflows/sca.yml:69]: > what's the downside of having it?

I was asking myself the opposite question: what is the benefit of having it? And could not find any.

Comment (keradus) [reply .github/workflows/sca.yml:69]: i know it's not frozen and progressing.
i was struggling with it a lot for slow tool like phpmd called for whole project at once, so i had to workaround it with checking only modified files. for phpstan we are checking whole codebase

**Comment by keradus** [comment]
I believe that discussion in this PR took more tham than actual value of having or not having this changes in.
Maybe let's maintain status quo and put focus on more important topics

**Pull request closed without merging**